pkg/info/infopb/rpc.proto

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+syntax = "proto3";
+package thanos;
+
+import "store/labelpb/types.proto";
+import "gogoproto/gogo.proto";
+
+option go_package = "infopb";
+
+option (gogoproto.sizer_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (gogoproto.goproto_getters_all) = false;
+
+// Do not generate XXX fields to reduce memory footprint and opening a door
+// for zero-copy casts to/from prometheus data types.
+option (gogoproto.goproto_unkeyed_all) = false;
+option (gogoproto.goproto_unrecognized_all) = false;
+option (gogoproto.goproto_sizecache_all) = false;
+
+// Info represents the API that is responsible for gathering metadata about the all APIs supported by the component.
+service Info {
+    // Info returns the metadata (Eg. LabelSets, Min/Max time) about all the APIs the component supports.
+    rpc Info(InfoRequest) returns (InfoResponse);
+}
+
+message InfoRequest {}
+
+message InfoResponse {
+    repeated ZLabelSet label_sets = 1 [(gogoproto.nullable) = false];
+    string ComponentType          = 2;
+    
+    StoresInfo stores             = 3;
```

Comment:
Also sorry for confusion, we suggested to check other APIs to make sure we the naming makes sense. We definitely have API called `Store`  not `Stores` so that one is wrong (`StoreInfo` makes more sense same with variable, there is single Store implementation there).

For others let's check but I see:

```
/// Rules represents API that is responsible for gathering rules and their statuses.
service Rules {
    /// Rules has info for all rules.
    /// Returned rules are expected to include external labels.
    rpc Rules(RulesRequest) returns (stream RulesResponse);
}
/// Targets represents API that is responsible for gathering targets and their states.
service Targets {
    /// Targets has info for all targets.
    /// Returned targets are expected to include external labels.
    rpc Targets (TargetsRequest) returns (stream TargetsResponse);
}
/// Exemplars represents API that is responsible for gathering exemplars and their states.
service Exemplars {
    /// Exemplars has info for all exemplars.
    /// Returned exemplars are expected to include external labels.
    rpc Exemplars(ExemplarsRequest) returns (stream ExemplarsResponse);
}
service Metadata {
  rpc MetricMetadata(MetricMetadataRequest) returns (stream MetricMetadataResponse);
}
```

I guess others then makes sense, so... can we revert to what we had? Sorry for confusion 🤦🏽

Reply:
No worries! Thank you for checking out APIs. I will revert back to `StoreInfo`